backend/compact-connect/stacks/feature_flag_stack/feature_flag_resource.py

```diff
@@ -0,0 +1,168 @@
+"""
+CDK construct for managing StatSig feature flags as custom resources.
+
+This construct creates a CloudFormation custom resource that manages the lifecycle
+of StatSig feature flags across different environments.
+"""
+
+import os
+from enum import StrEnum
+
+import jsii
+from aws_cdk import CustomResource, Duration, Stack
+from aws_cdk.aws_iam import IGrantable, PolicyStatement
+from aws_cdk.aws_logs import RetentionDays
+from aws_cdk.custom_resources import Provider
+from cdk_nag import NagSuppressions
+from common_constructs.python_function import PythonFunction
+from constructs import Construct
+
+
+class FeatureFlagEnvironmentName(StrEnum):
+    TEST = 'test'
+    BETA = 'beta'
+    PROD = 'prod'
+    # add sandbox environment names here if needed
+    SANDBOX = 'sandbox'
+
+
+@jsii.implements(IGrantable)
+class FeatureFlagResource(Construct):
+    """
+    Custom resource for managing StatSig feature flags.
+
+    This construct creates a Lambda-backed custom resource that handles
+    creation, updates, and deletion of feature flags in StatSig.
+    """
+
+    def __init__(
+        self,
+        scope: Construct,
+        construct_id: str,
+        *,
+        flag_name: str,
+        auto_enable_envs: list[FeatureFlagEnvironmentName],
+        custom_attributes: dict[str, str] | dict[str, list] | None = None,
+        environment_name: str,
+    ):
+        """
+        Initialize the FeatureFlagResource construct.
+
+        :param flag_name: Name of the feature flag to manage
+        :param auto_enable_envs: List of environments to automatically enable the flag for
+        :param custom_attributes: Optional custom attributes for feature flag targeting
+        :param environment_name: The environment name (test, beta, prod)
+        """
+        super().__init__(scope, construct_id)
+
+        if not flag_name or not environment_name:
+            raise ValueError('flag_name and environment_name are required')
+
+        # Lambda function for managing feature flags
+        self.manage_function = PythonFunction(
+            self,
+            'ManageFunction',
+            index=os.path.join('handlers', 'manage_feature_flag.py'),
+            lambda_dir='feature-flag',
+            handler='on_event',
+            log_retention=RetentionDays.ONE_MONTH,
+            environment={'ENVIRONMENT_NAME': environment_name},
+            timeout=Duration.minutes(5),
+            memory_size=256,
+        )
+
+        # Grant permissions to read secrets
+        secret_name = f'compact-connect/env/{environment_name}/statsig/credentials'
+        self.manage_function.add_to_role_policy(
```

Comment:
Did [Secret.from_secret_name_v1()](https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_secretsmanager/Secret.html#aws_cdk.aws_secretsmanager.Secret.from_secret_name_v2) not work for this?